File: CS/DXWebApplication1/Services/CustomWebDocumentViewerExceptionHandler.cs
using System;
using System.IO;
using System.ServiceModel;
using DevExpress.XtraReports.Web.WebDocumentViewer;

namespace DXWebApplication1.Services {
	public class CustomWebDocumentViewerExceptionHandler : WebDocumentViewerExceptionHandler {
		public override string GetDocumentCreationExceptionMessage(DocumentCreationException ex) {
			if (ex.InnerException.InnerException is MyCustomException) {
				return ex.InnerException.InnerException.Message;
			} else {
				return "NotImplementedException occured.";
			};
		}

		public override string GetFaultExceptionMessage(FaultException faultException) {
			return "FaultException occured.";
		}
	
		public override string GetUnknownExceptionMessage(Exception ex) {
			if (ex is FileNotFoundException)
				return "FileNotFoundException occured.";
			return ex.Message;
		}
	}
}

File: VB/DXWebApplication1/Services/CustomWebDocumentViewerExceptionHandler.vb
Imports System
Imports System.IO
Imports System.ServiceModel
Imports DevExpress.XtraReports.Web.WebDocumentViewer

Namespace DXWebApplication1.Services
	Public Class CustomWebDocumentViewerExceptionHandler
		Inherits WebDocumentViewerExceptionHandler

		Public Overrides Function GetDocumentCreationExceptionMessage(ByVal ex As DocumentCreationException) As String
			Dim customExpressionException = TryCast(ex.GetBaseException(), CustomInvalidRangeException)
			If customExpressionException IsNot Nothing Then
				Return customExpressionException.DetailedMessage
			Else
				Return MyBase.GetDocumentCreationExceptionMessage(ex)
			End If
		End Function

		Public Overrides Function GetFaultExceptionMessage(ByVal faultException As FaultException) As String
			Return "FaultException occurred: " & faultException.Message & "."
		End Function

		Public Overrides Function GetUnknownExceptionMessage(ByVal ex As Exception) As String
			If TypeOf ex Is FileNotFoundException Then
#If DEBUG Then
				Return "Debug mode. " & ex.Message & "."
#Else
				Return "File is not found."
#End If
			End If
			Return ex.GetType().Name & " occurred. See the log file for more details."
		End Function
	End Class
End Namespace